handlers/aws/sqs_trigger.py

```diff
@@ -4,35 +4,94 @@
+from utils import get_bucket_name_from_arn
 
+from share import Config, logger
 from storage import CommonStorage, StorageFactory
 
 
-def _handle_sqs_event(event) -> Generator[tuple[dict[str, any], int], None, None]:
-    for sqs_record in event["Records"]:
+def _handle_sqs_continuation(
+    sqs_continuing_queue: str,
+    lambda_event: dict[str, Any],
+    event_input_id: str,
+    last_ending_offset: int,
+    current_sqs_record: int,
+    current_s3_record: int,
+    config_yaml: str,
+):
+
+    sqs_records = lambda_event["Records"][current_sqs_record:]
+    body = json.loads(sqs_records[0]["body"])
+    body["Records"] = body["Records"][current_s3_record:]
+    body["Records"][0]["last_ending_offset"] = last_ending_offset + 1
+    sqs_records[0]["body"] = json.dumps(body)
+
+    sqs_client = boto3.client("sqs")
+    for sqs_record in sqs_records:
+        sqs_client.send_message(
+            QueueUrl=sqs_continuing_queue,
+            MessageBody=sqs_record["body"],
+            MessageAttributes={
+                "config": {"StringValue": config_yaml, "DataType": "String"},
+                "originalEventSource": {"StringValue": event_input_id, "DataType": "String"},
+            },
+        )
+
+        logger.debug("continuing", extra={"sqs_continuing_queue": sqs_continuing_queue, "body": body})
+
+
+def _handle_sqs_event(config: Config, event) -> Generator[tuple[dict[str, Any], int, int, int, int], None, None]:
+    for sqs_record_n, sqs_record in enumerate(event["Records"]):
+        event_input = config.get_input_by_type_and_id("sqs", sqs_record["eventSourceARN"])
+        if not event_input:
+            return None
+
         body = json.loads(sqs_record["body"])
-        for s3_record in body["Records"]:
+        for s3_record_n, s3_record in enumerate(body["Records"]):
             aws_region = s3_record["awsRegion"]
             bucket_arn = s3_record["s3"]["bucket"]["arn"]
             object_key = s3_record["s3"]["object"]["key"]
+            last_ending_offset = s3_record["last_ending_offset"] if "last_ending_offset" in s3_record else 0
 
             if len(bucket_arn) == 0 or len(object_key) == 0:
-                raise Exception("cannot find bucket_arn or object_key for s3")
+                raise Exception("Cannot find bucket_arn or object_key for s3")
 
-            bucket_name: str = _get_bucket_name_from_arn(bucket_arn)
+            bucket_name: str = get_bucket_name_from_arn(bucket_arn)
             storage: CommonStorage = StorageFactory.create(
                 storage_type="s3", bucket_name=bucket_name, object_key=object_key
             )
 
-            for log_event, offset in storage.get():
+            span = elasticapm.capture_span(f"WAIT FOR OFFSET STARTING AT {last_ending_offset}")
+            span.__enter__()
+            for log_event, beginning_offset, ending_offset in storage.get_by_lines(
+                range_start=last_ending_offset, last_ending_offset=last_ending_offset
+            ):
```

Comment:
Nit. Consider splitting long fors or if conditions with variables. Something like this here:
```suggestion
            events = storage.get_by_lines(
                range_start=last_ending_offset,
                last_ending_offset=last_ending_offset,
            )
            for log_event, beginning_offset, ending_offset in events:
```